### PR TITLE
[backbone-router] fix incorrect PBBR ALOC

### PR DESCRIPTION
**Pull request by simonlingoogle**

### CODE REVIEW

File: tests/scripts/thread-cert/test_rloc.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        pbbr.start()
+        self.simulator.go(10)
+        self.assertEqual('router', pbbr.get_state())
+        self.assertTrue(pbbr.is_primary_backbone_router)
+
+        router.start()
+        self.simulator.go(10)
+        self.assertEqual('router', router.get_state())
+
+        print(pbbr.get_addrs())
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/test_rloc.py:92]:
```suggestion
```

**Review by superwhd — APPROVED** [review]
LGTM :+1:

File: tests/scripts/thread-cert/test_rloc.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+import re
+import unittest
+
+import thread_cert
+from config import MESH_LOCAL_PREFIX_REGEX_PATTERN, ROUTING_LOCATOR_REGEX_PATTERN
+
+# Test description:
+# The purpose of this test is to verify ALOC functionalities.
```
Comment (librasungirl) [inline tests/scripts/thread-cert/test_rloc.py:36]: This PR fixes an incorrect construction of ALOC, while this test seems verifying the connectivity when dest is ALOC. What does "verify ALOC functionalities" mean here?

Comment (simonlingoogle) [inline tests/scripts/thread-cert/test_rloc.py:36]:
```suggestion
# The purpose of this test is to verify ALOC connectivity.
```

**Review by librasungirl — APPROVED** [review]

**Comment by jwhui** [comment]
@simonlingoogle , ALOC test seems to be failing. Can you take a look?

**Comment by simonlingoogle** [comment]
> @simonlingoogle , ALOC test seems to be failing. Can you take a look?

Yes. The new test `test_aloc.py` can not be executed in `Makefile check` because it's using both both normal simulation build and BBR simulation build. 

With #6508 , the new test can be automatically included in `packet-verfication` checks.

**Comment by jwhui** [comment]
> Yes. The new test `test_aloc.py` can not be executed in `Makefile check` because it's using both both normal simulation build and BBR simulation build.

Makes sense, thanks.

> With #6508 , the new test can be automatically included in `packet-verfication` checks.

Some tests are still failing, including `test_pbbr_aloc.py` - https://github.com/openthread/openthread/pull/6505/checks?check_run_id=2436310000

**Comment by simonlingoogle** [comment]
> Some tests are still failing, including `test_pbbr_aloc.py` - https://github.com/openthread/openthread/pull/6505/checks?check_run_id=2436310000

It's difficult to make this test runs in Thread 1.1  `packet verification` because one node is 1.2 BBR. I skip running this test for Thread 1.1 for now.

**Review by jwhui — APPROVED** [review]

**Pull request merged**